### PR TITLE
add sakiparens.py

### PR DESCRIPTION
**Pull request by SakiFu**

### CODE REVIEW

File: sakiparens.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+def parens(string):
+    stack = []
+    for char in string:
+        if char == '(':
+            stack.append(char)
+        elif char == ')':
+            try:
+                stack.pop()
+            except:
```
Comment (cewing) [inline sakiparens.py:11]: You probably want to catch a specific exception.  What is raised by the stack if `.pop()` is called on an empty stack?  That's the error you should be catching, not just any error.

File: sakiparens.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
+from __future__ import unicode_literals
+
+def parens(string):
+    stack = []
```
Comment (cewing) [inline sakiparens.py:4]: You should be using an actual stack here, not a Python list.

File: sakiparens.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    for char in string:
+        if char == '(':
+            stack.append(char)
+        elif char == ')':
+            try:
+                stack.pop()
```
Comment (cewing) [inline sakiparens.py:10]: `.pop()` on a list is an expensive operation (O(n)).  You can do much better by using an actual stack where both insert and pop are O(1)

File: sakiparens.py (hunk 1 above)
Comment (SakiFu) [inline sakiparens.py:11]: Sure I meant IndexError. I'will fix it.

File: sakiparens.py (hunk 3 above)
Comment (ndraper2) [inline sakiparens.py:10]: @cewing I thought popping from the end of a list ran in constant time, and that pop(0) was horrendous?

Comment (cewing) [inline sakiparens.py:10]: dope-slap for me.  you are right.

**Pull request merged**